java/arcs/core/analysis/Deduction.kt

```diff
@@ -0,0 +1,99 @@
+package arcs.core.analysis
+
+import arcs.core.data.AccessPath
+import arcs.core.data.Claim
+import arcs.core.data.expression.Expression
+
+/** Field [Identifier]. */
+typealias Identifier = String
+
+/** Lists of [Identifier]s imply an [AccessPath]. */
+typealias Path = List<Identifier>
+
+/**
+ * [Deduction]s represent how inputs of a Paxel [Expression] contribute to its outputs.
+ *
+ * [Deduction]s recursively associate identifiers to field paths. These relationships can be
+ * directly translated into [Claim]s and [AccessPath]s.
+ */
+sealed class Deduction {
+
+  /** Substitute aliases in all paths. */
+  abstract fun substitute(aliases: Scope): Deduction
+
+  /** Union of two [Deduction] collections. */
+  abstract operator fun plus(other: Deduction): Deduction
+
+  /** A representation of a lookup claim in a Paxel [Expression]. */
+  data class Equal(val path: Path = emptyList()) : Deduction() {
+
+    constructor(vararg paths: Identifier) : this(paths.toList())
```

Comment:
`paths` is already a list. Isn't `toList` redundant?

Reply:
I think `paths` is an Array, but I'll double check.

Reply:
Yes, `toList` here seems to be required.